### PR TITLE
Added `databricks_service_principal_secret` documentation

### PR DESCRIPTION
**Pull request by jackyhu-db**
## Changes
- Add the resource `databricks_service_principal_secret` into the `docs/resources`
- Add the section `Authenticating with hostname, client_id and client_secret` in the `docs/index.md`

## Tests

- [ ] `make test` run locally
- [x] relevant change in `docs/` folder
- [ ] covered with integration tests in `internal/acceptance`
- [ ] relevant acceptance tests are passing
- [ ] using Go SDK

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -172,6 +173,48 @@ provider "databricks" {
 }
 ```
 
+### Authenticating with hostname, client_id and client_secret
```
Comment (alexott) [inline docs/index.md:176]: Should we include a separate section for AWS, like we have for Azure & GCP?

```
## Special configurations for AWS
```

File: docs/resources/service_principal_secret.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+---
+# databricks_service_principal_service Resource
+
+With this resource you can create a secret under given [Service Principals](https://docs.databricks.com/administration-guide/users-groups/service-principals.html)
+
+With the secret you can configure Databricks Terraform Provider to authenticate with service principal UUID and secret. See [Authenticating with service principal ID and secret](../index.md#authenticating-with-hostname-client_id-and-client_secret).
+
+You can also use the secret to request OAuth tokens for the service prinicipal, which can be used to authenticate to Databricks APIs. See [Authentication using OAuth tokens for service principals](https://docs.databricks.com/dev-tools/api/latest/authentication-oauth.html).
+
+-> **Note** This resource in only available in AWS.
```
Comment (alexott) [inline docs/resources/service_principal_secret.md:12]: Move this note closer to the top of the file

File: docs/resources/service_principal_secret.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+## Example Usage
+
+Create service principal secret
+
+```hcl
+resource "databricks_service_principal_secret" "terraform_sp" {
+  service_principal_id = "123456789012345"
```
Comment (alexott) [inline docs/resources/service_principal_secret.md:20]: if we say UUID, let specify it in the correct format

**Review by alexott — CHANGES_REQUESTED** [review]

File: docs/resources/service_principal_secret.md (hunk 3 above)
Comment (jackyhu-db) [reply docs/resources/service_principal_secret.md:20]: Actually service principal ID and UUID are different. This is the correct format of the ID.

**Review by bryanmcq-db — APPROVED** [review]
looks good, thanks!

File: docs/resources/service_principal_secret.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+Additionally, the secret can be used to request OAuth tokens for the service principal, which can be used to authenticate to Databricks REST APIs. See [Authentication using OAuth tokens for service principals](https://docs.databricks.com/dev-tools/authentication-oauth.html).
+
+
+## Example Usage
+
+Create service principal secret
+
+```hcl
+resource "databricks_service_principal_secret" "terraform_sp" {
+  service_principal_id = "123456789012345"
```
Comment (alexott) [inline docs/resources/service_principal_secret.md:21]: we can write this as to show that it refer to existing service principal:

```
service_principal_id = databricks_service_principal.this.id
```

**Review by alexott — APPROVED** [review]
looks good, just one comment that we can add later

**Review by mgyucht — APPROVED** [review]

File: docs/resources/service_principal_secret.md (hunk 4 above)
Comment (nfx) [inline docs/resources/service_principal_secret.md:21]:
```suggestion
  service_principal_id = databricks_service_principal.this.id
```

**Review by nfx — APPROVED** [review]

**Pull request merged**